iphone/CoreApi/CoreApi/Traffic/MWMMapOverlayManager.mm

```diff
@@ -144,54 +154,39 @@ + (BOOL)outdoorEnabled
   return self.outdoorState != MWMMapOverlayOutdoorStateDisabled;
 }
 
-+ (void)setTrafficEnabled:(BOOL)enable
++ (BOOL)hikingEnabled
 {
-  if (enable)
-  {
-    [self setTransitEnabled:false];
-    [self setIsoLinesEnabled:false];
-  }
+  return self.hikingState != MWMMapOverlayHikingStateDisabled;
+}
 
++ (BOOL)cyclingEnabled
+{
+  return self.cyclingState != MWMMapOverlayCyclingStateDisabled;
+}
+
++ (void)setTrafficEnabled:(BOOL)enable
+{
   auto & f = GetFramework();
   f.GetTrafficManager().SetEnabled(enable);
   f.SaveTrafficEnabled(enable);
 }
 
 + (void)setTransitEnabled:(BOOL)enable
 {
-  if (enable)
-  {
-    [self setTrafficEnabled:!enable];
-    [self setIsoLinesEnabled:false];
-    [self setOutdoorEnabled:false];
-  }
-
   auto & f = GetFramework();
   f.GetTransitManager().EnableTransitSchemeMode(enable);
   f.SaveTransitSchemeEnabled(enable);
 }
 
 + (void)setIsoLinesEnabled:(BOOL)enable
 {
-  if (enable)
-  {
-    [self setTrafficEnabled:false];
```

Comment:
Would it be too slow when all layers are enabled at the same time? What is the use-case?

Reply:
Made the same logic with Android. I don't see any reason to limit the combination of layers. The outdoor layer shows pedestrian infra better, and combining with subways is ok.